docs/deprecation.md

```diff
@@ -0,0 +1,23 @@
+# Deprecation Policy
+
+## User-facing commands and their arguments
+
+Whenever you rename a command or make a previously optional argument required,
+we require that you preserve the old command invocations keeps on working for 6
+months (so 6 releases, since we release monthly) with a deprecation message.
+The message should inform the user that the previous workflow is deprecated
+and to be removed in the future.
+
+## Packaging commands and niche commands
```

Comment:
(For discussion) You don't really discuss packaging commands in here. I think it'd make sense to have a separate section about how we treat packager-facing deprecations, but I'm not sure what it should say.

Reply:
cc @emilazy since she brought it up in the [Discord thread]. I'm also happy to combine it with the next section if that works for you. 

[Discord thread]: https://discord.com/channels/968932220549103686/1334974861067686004

Reply:
I renamed it to just "Niche dommands", maybe that fits better?